### PR TITLE
Refactor jaxrs and OpenAPI support

### LINKED ISSUE #5343: Weird "default" answer from the external API?
In #5308 we saw a weird answer from a newer external API endpoint when you query it in a certain way:

```json
{
  "available": []
}
```

This response doesn't really seem to have anything to do with that specific endpoint, so I dug around a bit and indeed, you can also get it from other, totally unrelated endpoints, for example

```sh
curl -v -L 'https://develop.opencast.org/api/workflows/1.json' \
  -H 'accept: application/v1.0.0+json' \
  -u admin:opencast
```

Two things to note about that call:

* [The endpoint](https://develop.opencast.org/docs.html?path=/api/workflows#getworkflowinstance-1) doesn't actually end in `.json`,
* and we are querying a version which it is not yet part of.

With both of these, that should also give you the weird response. It seems to be some kind of fallback response? But it's not very useful at that, so I don't think it's intended?

### PR DESCRIPTION
**Pull request by geichelberger**
### What's new:
- `/openapi-ui` - Swagger-UI
- `/openapi.(json|yaml)`

This replaces the old servlet resource comparator method with a jaxrs whiteboard-compatible implementation. 
Furthermore, it supports OpenAPI/Swagger-UI via the CXF OpenAPI feature.

This change should also fix our APIs, which return entirely unrelated data. e.g. https://develop.opencast.org/api/admin-ng/workflows

### BREAKING CHANGE
It's not possible anymore to serve the admin-ng interface and the admin-ng API on the same path. ~~This PR moved it to `/admin-ng-ui`.~~ The better choice would be to rename the Admin UI API endpoints from admin-ng to admin-api as that change is not user-facing and the API is only internal. 

Fixes https://github.com/opencast/opencast/issues/5343

### TODO:

* [x] Fix  a little bug caused by caching, which can result in multiple errors when requesting `/openapi.json`
` java.lang.IllegalArgumentException: Cannot use a ScanResult after it has been closed`
* [ ] Update documentation and add openapi-ui or replace rest_docs links.
* [x] WelcomeResource `opencast.svg` wrong content type.
* [ ] ~~New Legacy Admin UI home or API home.~~
* [ ] Test Opencast Cluster

Custom Version:
grafik

Original Version:
grafik

* [ ] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by KatrinIhler** [comment]
FYI: Branch cut for OC 17 is on November 6!

**Comment by geichelberger** [comment]
While rebasing, I found a few missing endpoints; I hope I have now annotated all endpoints with @JaxrsResource. Searching with IntelliJ is not reliable.

**Comment by geichelberger** [comment]
I rebased it and replaced the default swagger-ui with a customized one.

File: modules/transcription-service-microsoft-azure/pom.xml
Comment (gregorydlogan) [inline modules/transcription-service-microsoft-azure/pom.xml:1]: What the heck happened here?

**Review by gregorydlogan — CHANGES_REQUESTED** [review]
Looks good.  Single file which probably got reformatted accidentally.  Found a bunch more `@Path("/")` annotations with `grep`.  A bunch are in test classes and thus can *probably* be ignored, but there are a few not-test ones:
```
greg@fw13:~/opencast/upstream$ grep -R 'Path("/")'
modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/endpoint/WowzaStreamingDistributionRestService.java:  @Path("/")
modules/runtime-info-ui/src/main/java/org/opencastproject/runtimeinfo/ui/WelcomeResources.java:@Path("/")
modules/publication-service-oaipmh-remote/src/test/java/org/opencastproject/publication/oaipmh/endpoint/TestOaiPmhPublicationRestService.java:@Path("/")
modules/publication-service-youtube-v3/src/main/java/org/opencastproject/publication/youtube/endpoint/YouTubePublicationRestService.java:  @Path("/")
modules/distribution-service-download/src/main/java/org/opencastproject/distribution/download/endpoint/DownloadDistributionRestService.java:  @Path("/")
modules/metrics-exporter/src/main/java/org/opencastproject/metrics/impl/MetricsExporter.java:  @Path("/")
modules/distribution-service-aws-s3/src/main/java/org/opencastproject/distribution/aws/s3/endpoint/AwsS3DistributionRestService.java:  @Path("/")
modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/UsersEndpoint.java:  @Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestSeriesEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestListProvidersEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestThemesEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestJobEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestServicesEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestEventEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestTasksEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestAclEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestUsersEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestUserSettingsEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestServerEndpoint.java:@Path("/")
modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/TestCaptureAgentsEndpoint.java:@Path("/")
modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/endpoint/IncidentServiceEndpoint.java:  @Path("/")
modules/lti/src/main/java/org/opencastproject/lti/LtiServiceGuiEndpoint.java:  @Path("/")
modules/lti-service-impl/src/main/java/org/opencastproject/lti/service/endpoint/LtiServiceRestEndpoint.java:  @Path("/")
modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java:  @Path("/")
modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java:  @Path("/")
modules/external-api/src/main/java/org/opencastproject/external/endpoint/WorkflowDefinitionsEndpoint.java:  @Path("/")
modules/external-api/src/main/java/org/opencastproject/external/endpoint/CaptureAgentsEndpoint.java:  @Path("/")
modules/external-api/src/test/java/org/opencastproject/external/endpoint/TestEventsEndpoint.java:@Path("/")
modules/external-api/src/test/java/org/opencastproject/external/endpoint/TestBaseEndpoint.java:@Path("/")
modules/external-api/src/test/java/org/opencastproject/external/endpoint/TestWorkflowDefinitionsEndpoint.java:@Path("/")
modules/external-api/src/test/java/org/opencastproject/external/endpoint/TestWorkflowsEndpoint.java:@Path("/")
modules/external-api/src/test/java/org/opencastproject/external/endpoint/TestCaptureAgentsEndpoint.java:@Path("/")
modules/kernel/src/test/java/org/opencastproject/kernel/bundleinfo/TestBundleInfoRestEndpoint.java:@Path("/")
modules/series-service-impl/src/main/java/org/opencastproject/series/endpoint/SeriesRestService.java:  @Path("/")
modules/userdirectory/src/main/java/org/opencastproject/userdirectory/endpoint/UserEndpoint.java:  @Path("/")
modules/oaipmh/src/test/java/org/opencastproject/oaipmh/server/TestRestService.java:@Path("/")
modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/endpoint/SchedulerRestService.java:  @Path("/")
modules/publication-service-oaipmh/src/main/java/org/opencastproject/publication/oaipmh/endpoint/OaiPmhPublicationRestService.java:  @Path("/")
```

Tested with both standalone, and clustered configurations.  No obvious explosions, so IMO this should go in as soon as the last few endpoints get done.

File: modules/transcription-service-microsoft-azure/pom.xml
Comment (geichelberger) [reply modules/transcription-service-microsoft-azure/pom.xml:1]: The last merge conflict caused this; the transcription-azure PR changed the indent of the pom.xml, and I fixed it and applied the old indent.

**Comment by geichelberger** [comment]
The Paths, except for the test cases, are not service root paths. The annotation should be between lines 60 and 90 if it is a class annotation. I did some samples, and they were all false positives.

My changes are replacing the path `/openapi-ui` with `/rest-docs` and adding a security xml entry. The new rest-docs now require a logged-in user.

**Pull request merged**